src/logger/MetricsContext.ts

```diff
@@ -107,14 +108,46 @@ export class MetricsContext {
   }
 
   /**
-   * Validates dimension set length is not more than Constants.MAX_DIMENSION_SET_SIZE
+   * Validates dimension set.
+   * @see [CloudWatch Dimensions](https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_Dimension.html)
    *
    * @param dimensionSet
    */
   public static validateDimensionSet(dimensionSet: Record<string, string>): void {
     if (Object.keys(dimensionSet).length > Constants.MAX_DIMENSION_SET_SIZE)
       throw new DimensionSetExceededError(
-        `Maximum number of dimensions per dimension set allowed are ${Constants.MAX_DIMENSION_SET_SIZE}`)
+        `Maximum number of dimensions per dimension set allowed are ${Constants.MAX_DIMENSION_SET_SIZE}`,
+      );
+
+    // Validate dimension key and value are valid strings
+    Object.entries(dimensionSet).forEach(([key, value]) => {
+      dimensionSet[key] = String(value);
+
+      if (
+        !MetricsContext.isAscii(key) ||
+        key.trim().length == 0 ||
+        key.startsWith(':') ||
+        key.length > Constants.MAX_DIMENSION_NAME_LENGTH
+      ) {
+        throw new InvalidDimensionError(`Dimension value ${value} is invalid`);
+      }
+
+      if (
+        !MetricsContext.isAscii(value) ||
+        value.trim().length == 0 ||
+        value.length > Constants.MAX_DIMENSION_VALUE_LENGTH
+      ) {
+        throw new InvalidDimensionError(`Dimension value ${value} is invalid`);
+      }
+    });
+  }
+
+  /**
+   * Check if the string contains only ASCII characters.
+   * @param str string to check
+   */
+  public static isAscii(str: string): boolean {
+    return /^[\x20-\x7F]*$/.test(str);
```

Comment:
Out of curiosity, have we checked if there's a native Node.js way for checking this? Or perhaps a widely used library that can do this check for us?

Reply:
I could not find a native Node.js way or any widely used library that can do this for us.

Reply:
Not much of a better suggestion, but I believe latest JS (2018+) has this regex that specifically checks for ASCII characters: ```/[\p{ASCII}]+/u``` ([test](https://regex101.com/r/gGjNmr/1)). Not sure if its applicable here. 
Personally, I don't think using regex is a bad option though as it is relatively simple enough and future proof (js string is always in UTF-16 I believe?)

Reply:
I agree that it is relatively future proof, but we could always use [validator](https://www.npmjs.com/package/validator) to check for ASCII.